merch/gen/printfulutils.py

```diff
@@ -0,0 +1,346 @@
+import requests
+import json
+import errno
+import logging
+import os
+import requests
+import sentry_sdk
+import struct
+import subprocess
+import codecs
+from datadog import initialize, api, statsd
+from urllib.parse import urljoin
+from base64 import standard_b64encode, b64encode
+from __future__ import unicode_literals
+from functools import partialmethod
+from printful._compat import urllib, force_text, force_bytes
+from sentry_sdk.integrations.logging import LoggingIntegration
+from base64 import b64encode
+
+class PrintfulException(Exception):
+    """ Used for response JSON validation """
+
+
+class PrintfulAPIException(Exception):
+    """ Invalid API Server Responses
+    """
+    def __init__(self, resp):
+        self.code = resp['code']
+        self.result = resp['result']
+
+    def __str__(self):
+        return 'Server Response {0}: {1}'.format(self.code,
+                                          self.result)
+
+
+class Printful:
+
+    def __init__(self, key):
+        """
+        This is the main client that handles all of the requests
+        to the Printful API.  The class uses [requests](http://docs.python-requests.org/en/latest/)
+        as a way to interact with the server.
+
+        :param key: API key generated by Printful
+
+        Usage::
+
+            >>> import printful
+            >>> pf = printful.Printful(key)
+            >>> orders = pf.get('orders')
+
+        """
+        self.key = bytearray(key, 'utf-8')
+        self.api_url = 'https://api.theprintful.com/'
+        self.user_agent = 'Printful API Python Library 1.2'
+        self._response = {}
```

Comment:
Please consider relying on an existing package such as [PyPrintful](https://pypi.org/project/PyPrintful/) instead of copying a sample wrapper from documentation.

As mentioned previously, why are we using this API?